stdlib/websub/src/main/ballerina/websub/h2_hub_persistence.bal

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/h2;
+import ballerina/log;
+
+const string CREATE_TOPICS_TABLE = "CREATE TABLE IF NOT EXISTS topics (topic VARCHAR(255), PRIMARY KEY (topic))";
+const string INSERT_INTO_TOPICS = "INSERT INTO topics (topic) VALUES (?)";
+const string DELETE_FROM_TOPICS = "DELETE FROM topics WHERE topic=?";
+const string SELECT_ALL_FROM_TOPICS = "SELECT * FROM topics";
+
+const string CREATE_SUBSCRIPTIONS_TABLE = "CREATE TABLE IF NOT EXISTS subscriptions
+                                            (topic VARCHAR(255), callback VARCHAR(255), secret VARCHAR(255),
+                                            lease_seconds BIGINT, created_at BIGINT, PRIMARY KEY (topic, callback))";
+const string INSERT_INTO_SUBSCRIPTIONS_TABLE = "INSERT INTO subscriptions
+                                                (topic,callback,secret,lease_seconds,created_at) VALUES (?,?,?,?,?)";
+const string DELETE_FROM_SUBSCRIPTIONS = "DELETE FROM subscriptions WHERE topic=? AND callback=?";
+const string SELECT_FROM_SUBSCRIPTIONS = "SELECT topic, callback, secret, lease_seconds, created_at FROM subscriptions";
+
+# Represents H2 based hub persistence configuration and functions.
+public type H2HubPersistenceObject object {
+
+    // TODO: make private
+    public h2:Client subscriptionDbClient;
+
+    public function __init(h2:Client subscriptionDbClient) {
+        self.subscriptionDbClient = subscriptionDbClient;
+        var ret = self.subscriptionDbClient->update(CREATE_TOPICS_TABLE);
+        if (ret is error) {
+            log:printError("'topics' table creation failed: " + <string>ret.detail().message);
+        }
+
+        ret = self.subscriptionDbClient->update(CREATE_SUBSCRIPTIONS_TABLE);
```

Comment:
Should we still continue creating other tables if previous table creation fails?